### PR TITLE
Q17-18 or something

### PR DESCRIPTION
**Pull request by ziuq**
More protection and celebratory elimination of some excessive quoting that was in my path.
We should be installing `ProhibitInterpolationOfLiteral` as this may be all the violations.

Some of my duplicated work, I would have resolved in reverse if I considered that it would be easier to review.  Sorry, I'll try to keep it in mind in the future.

### CODE REVIEW

File: lib/LedgerSMB.pm [hunk 1]
```diff
@@ -495,7 +495,7 @@ sub merge {
         }
         elsif ( !defined $dst_arg && !defined $src->{$arg} )
         {
-            $logger->trace("LedgerSMB.pm: merge setting \$dst_arg is undefined \$src->{\$arg} is undefined");
+            $logger->trace('LedgerSMB.pm: merge setting \$dst_arg is undefined \$src->{\$arg} is undefined');
```
Comment (ehuelsmann) [inline lib/LedgerSMB.pm]: Now that interpolation doesn't take place anymore, shouldn't the backslashes be removed?

File: lib/LedgerSMB/DBH.pm [hunk 2]
```diff
@@ -38,11 +38,11 @@ sub connect {
     return undef unless $username && $password;
     return undef if $username eq 'logout';
 
-    my $dbh = DBI->connect(qq|dbi:Pg:dbname="$company"|, $username, $password,
+    my $dbh = DBI->connect("dbi:Pg:dbname=$company", $username, $password,
            { PrintError => 0, AutoCommit => 0,
              pg_enable_utf8 => 1, pg_server_prepare => 0 })
         or return undef;
-    $dbh->do("set client_min_messages = 'warning'");
+    $dbh->do('set client_min_messages = \'warning\'');
```
Comment (ehuelsmann) [inline lib/LedgerSMB/DBH.pm]: Would be nice to use `q{}` here due to the escaping backslashes in the current solution.

File: lib/LedgerSMB/Scripts/payment.pm [hunk 3]
```diff
@@ -1406,7 +1406,7 @@ my %amount_to_be_used;
 my %ovp_repeated_invoices;
 my %invoice_id_amount_to_pay;
 my $count;
-my $warning = $Payment->{"warning"};
+my $warning = $Payment->{'warning'};
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Scripts/payment.pm]: below, you decided to leave `warning` unquoted. I agree with that choice, but then this one should be unquoted as well -- for consistency.

**Comment by ehuelsmann** [comment]
I love how this PR makes payment.pm much more idiomatic.

File: lib/LedgerSMB.pm (hunk 1 above)
Comment (ziuq) [reply lib/LedgerSMB.pm]: Yes.  Funny, I remember seeing this line and pausing and moving on.  I'm fairly confident that it is the only line with escaped varnames I touched in all this de-interpolation.

Correction pushed.

**Pull request merged**